pyt/cfg/stmt_visitor.py

```diff
@@ -555,23 +555,44 @@ def visit_For(self, node):
             path=self.filenames[-1]
         ))
 
-        if isinstance(node.iter, ast.Call) and get_call_names_as_string(node.iter.func) in self.function_names:
-            last_node = self.visit(node.iter)
-            last_node.connect(for_node)
+        self.process_loop_funcs(node.iter, for_node)
 
         return self.loop_node_skeleton(for_node, node)
 
+    def process_loop_funcs(self, comp_n, loop_node):
+        """
+        If the loop test node contains function calls, it connects the loop node to the nodes of
+        those function calls.
+
+        :param comp_n: The test node of a loop that may contain functions.
+        :param loop_node: The loop node itself to connect to the new function nodes if any
+        :return: None
+        """
+        if isinstance(comp_n, ast.Call) and get_call_names_as_string(comp_n.func) in self.function_names:
+            last_node = self.visit(comp_n)
+            last_node.connect(loop_node)
+
     def visit_While(self, node):
         label_visitor = LabelVisitor()
-        label_visitor.visit(node.test)
+        test = node.test  # the test condition of the while loop
+        label_visitor.visit(test)
 
-        test = self.append_node(Node(
+        while_node = self.append_node(Node(
             'while ' + label_visitor.result + ':',
             node,
             path=self.filenames[-1]
         ))
 
-        return self.loop_node_skeleton(test, node)
+        if isinstance(test, ast.Compare):
+            comparators = test.comparators
+            comparators.append(test.left)  # quirk. See https://greentreesnakes.readthedocs.io/en/latest/nodes.html#Compare
```

Comment:
I'm not 100% sure, but I'm worried that this will mutate `node.test.comparators` which might mess stuff up later on if you try to operate on `while_node.node`.

```suggestion
            self.process_loop_funcs(test.left)  # quirk. See https://greentreesnakes.readthedocs.io/en/latest/nodes.html#Compare
            for comparator in test.comparators:
                self.process_loop_funcs(comparator, while_node)
```

Reply:
You're correct, this mutates the original node. I'll clone the list.